### DISCUSSION TITLE
Simple split tabbed/ split stacked script

### DISCUSSION THREAD

**Discussion by RayZ0rr** [i3/i3#4905] [Show and tell]
## WHAT

This is a simple script to change the current container to a split container with tabbed/stacked layout. This is similar to `splith` or `splitv` commands. I call this simple because I haven't looked at all usecases like nested containers within other tabbed containers, scratchpads etc.

## WHY?

Split containers with horizontal or vertical orientation has an indicator color on border. However, there is nothing similar for `tabbed/stacked` layout or even a way to create split container with such a layout. One can chain commands like this: 
`# bindsym $mod+t split h, layout tabbed`  
but without an indicator of sort it may be pressed again and repeating this command will create nested split containers, making things complicated imo.(you can press the above keybind multiple times and use `focus parent` to test)

## Usage

Put this in i3 config file

    # Layout Manager
    set $i3_scripts ~/.config/i3/scripts
    bindsym $mod+t exec $i3_scripts/i3-deck.sh tabbed
    bindsym $mod+s exec $i3_scripts/i3-deck.sh stacked

Don't forget to make the script executable.

## Dependencies

1. jq

## Script

Save following file in `~/.config/i3/scripts/i3-deck.sh` and make it executable

```bash
#!/usr/bin/env bash

# Author: RayZ0rr (github.com/RayZ0rr)
#
# Dependencies:
# - jq     : To get container layout

current_layout() {
  info=$(i3-msg -t get_tree | jq -r 'recurse(.nodes[];.nodes!=null)|select(.nodes[].focused).layout')
  echo "${info}"
}

check=$(current_layout)

split_tabbed() {
  if [[ "$check" == "tabbed" ]] ; then
    i3-msg 'layout tabbed'
    exit 0
  elif [[ "$check" == "splith" ]] ; then
    i3-msg 'split h, layout tabbed'
    exit 0
  elif [[ "$check" == "splitv" ]] ; then
    i3-msg 'split v, layout tabbed'
    exit 0
  else
    printf "Scratchpad or Invalid layout\n----------------------------------------------\n"
    exit 1
  fi
}

split_stacking() {
  if [[ "$check" == "stacked" ]] ; then
    i3-msg 'layout stacking'
    exit 0
  elif [[ "$check" == "splith" ]] ; then
    i3-msg 'split h, layout stacking'
    exit 0
  elif [[ "$check" == "splitv" ]] ; then
    i3-msg 'split v, layout stacking'
    exit 0
  else
    printf "Scratchpad or Invalid layout\n----------------------------------------------\n"
    exit 1
  fi
}

case "${1}" in
  "tabbed")
    split_tabbed
    exit 0 ;;
  "stacking")
    split_stacking
    exit 0 ;;
  *)
    echo "Run 'i3-deck.sh [tabbed|stacking]' to create split container with tabbed or stacking layout"
    exit 1 ;;
esac
```

## Use and extend !!

**Comment by adamws** [comment]
thanks @RayZ0rr, I've made this simple tool https://github.com/adamws/i3-stack based on this idea
